internal/services/network/private_endpoint_resource.go

```diff
@@ -281,6 +283,17 @@ func resourcePrivateEndpointCreate(d *pluginsdk.ResourceData, meta interface{})
 		Tags: tags.Expand(d.Get("tags").(map[string]interface{})),
 	}
 
+	cosmosDbResIds := getCosmosDbResIdInPrivateServiceConnections(parameters.PrivateEndpointProperties)
+	for _, cosmosDbResId := range cosmosDbResIds {
+		log.Printf("[DEBUG] Add Lock For Private Endpoint %q, lock name: %q", id.Name, cosmosDbResId)
+		locks.ByName(cosmosDbResId, "azurerm_private_endpoint")
+	}
```

Comment:
could this not be
```suggestion
	for _, cosmosDbResId := range cosmosDbResIds {
		log.Printf("[DEBUG] Add Lock For Private Endpoint %q, lock name: %q", id.Name, cosmosDbResId)
		locks.ByName(cosmosDbResId, "azurerm_private_endpoint")
		defer locks.UnlockByName(cosmosDbResId, "azurerm_private_endpoint")
	}
```

Reply:
Hi @katbyte , I've tried defer in for loop and it would cause an IDE warning. Though we know that a for loop in defer is equal to defer in for, but I think the warning is annoying and want to avoid that. I'm ok with both implementations, which one do you prefer?
By the way, during my test there was a random error from time to time, i doubt there is some async logic on service side that could cause an error randomly and I'm querying the service team, and got no response yet. Once I've solved it I'll submit this pr formally.

Reply:
what IDE warning do you get?

and sounds good

Reply:
JetBrains Goland would warn "Possible resource leak, 'defer' is called in the 'for' loop 
 Inspection info: Reports defer statements inside loops.
Using defer in loops can lead to resource leaks or unpredictable execution order of the statements."

I think it just want to avoid defer in a possibly infinity loop that caused resource leak or even worse a stackoverflow.

Reply:
And here's JetBrains's [link](https://www.jetbrains.com/help/go/code-inspections-in-go.html#Control_flow_issues).

Reply:
its a valid warning however i don't think it applies in this case as we don't care about the order of the defer statements as long as they are called? - also as long as golang-ci doesn't flag it it should be ok?

Reply:
Roger that.